controllers/client_policy.go

```diff
@@ -206,28 +226,50 @@ func (r *SingleClusterReconciler) appendCACertFromSecret(
 
 	found := &corev1.Secret{}
 
-	secretName := namespacedSecret(secretSource, defaultNamespace)
-	if err := r.Client.Get(context.TODO(), secretName, found); err != nil {
-		r.Log.Error(
-			err,
-			"Failed to get secret certificates to the pool, returning empty certPool",
-			"secret", secretName,
-		)
+	if secretSource.CaCertPath != nil {
+		secretName := namespacedSecret(secretSource.CaCertPath.SecretNamespace,
+			secretSource.CaCertPath.SecretName, defaultNamespace)
+		if err := r.Client.Get(context.TODO(), secretName, found); err != nil {
```

Comment:
I think we should return an error if there is an error in getting the secret.

Reply:
as per the discussion, will address it in next PR.